pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -440,11 +439,8 @@ def __validate_init_configuration(self) -> None:
                     " will duplicate the last checkpoint saved."
                 )
 
-    def __init_ckpt_dir(self, dirpath: Optional[Union[str, Path]], filename: Optional[str]) -> None:
-        self._fs = get_filesystem(str(dirpath) if dirpath else "")
-
-        if self.save_top_k != 0 and dirpath is not None and self._fs.isdir(dirpath) and len(self._fs.ls(dirpath)) > 0:
-            rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
```

Comment:
note for other reviewers:

This PR is an optimization we'd like to make when using remote filesystems with lightning. this code is currently called on every rank, which creates duplicate, bursty outbound network calls. when training with larger configurations, we ran into quota limitations for jobs our jobs. this will likely benefit anyone saving checkpoints using S3/GCS/azure storage

in reality, we only needed to check this on rank zero (assuming we have a common `dirpath` across nodes)